### ISSUE TITLE
PATCH of users/me route, changing the user's email, does not change the user's verification status

### ISSUE THREAD

**Issue by jakemanger** [fastapi-users/fastapi-users#778]
## Describe the bug

If I verify a user, then use the user's account to PATCH their user email via the users/me route, this does not change the user's verification status. 

I would think this would be a big problem in terms of security, as a user could sign up with a burner email, get verified and then change their email willy-nilly to anyone else and stay verified. They would then be able to claim that account (if the email hadn't already been used of course).

## Expected behavior

I think the expected behaviour would be that the verification status should be set to false whenever an email is changed.

I'm happy to implement this, but would need a bit of guidance where to get started.

**Comment by frankie567** [comment]
Yes, you're completely right. To be honest, I thought this was already covered 😅

Happy to review a PR about it! Basically, here is what would be needed:

1. Set the `is_verified` flag to `False` when updating e-mail. I wouls happen in the `_update` method of the user manager, right after line 547:

https://github.com/fastapi-users/fastapi-users/blob/f4818ee18ab07b79f079f9f3806975971e6995e0/fastapi_users/manager.py#L542-L547

2. Add a unit test for this so that we are sure we preserve this behavior. We could duplicate the following test for example:

https://github.com/fastapi-users/fastapi-users/blob/f4818ee18ab07b79f079f9f3806975971e6995e0/tests/test_router_users.py#L187-L203

And use `verified_user` instead of `user`. This way, we can check that, if we update the e-mail of `verified_user`, the `is_verified` flag is set to `False`.
